src/Analysis.cpp

```diff
@@ -580,11 +592,64 @@ void CodeAnalyzer::doAnalysis(Function &F, std::string optLoadFileName) {
     }
 }
 
+// assumes utilFunctionTester is actually an extended class and not UtilFunctionTester
+// (e.g., an instance of VariableTester)
```

Comment:
You could enforce this invariant with a `dynamic_cast`, if you think it's important.

Reply:
I tried even passing an instance of UtilFunctionTester but that is not possible since it's an abstract class, so I'll just update the comment to be more precise; that is, updating it to say "utilFunctionTester is an extended class of UtilFunctionTester"